skops/io/_dispatch.py

```diff
@@ -2,19 +2,34 @@
 
 import json
 
+from skops.io._utils import LoadState
+
 GET_INSTANCE_MAPPING = {}  # type: ignore
 
 
-def get_instance(state, src):
+def get_instance(state, load_state: LoadState):
     """Create instance based on the state, using json if possible"""
+
+    saved_id = state.get("__id__")
+    if saved_id and saved_id in load_state.memo:
```

Comment:
I wonder if we even need to check if the `"__id__"` key exists? I see that its presence is not strictly necessary for loading to work. On the other hand, will it ever be the case that the key doesn't exist if the `state` was created with skops? (Okay, if someone already used `dump` with the current main branch, but I think that's negligible?)

Overall, I'm fine either way, just considering if we can slightly simplify the code.

Reply:
I wanted to include that for backwards compatibility mostly, happy to remove these checks if we don't need backwards compatibility for models saved with 0.1 & 0.2 versions

Reply:
We added persistence after the 0.2 release, so it's not officially released at all. Therefore, I think we don't need to worry about it.

Reply:
Oh, I see, that makes sense. I can definitely remove this check then 👍